packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -69,6 +107,19 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
             Lib_ExecutionManagerWrapper.ovmCHAINID()
         );
 
+        // Compute the intrinsic gas
+        uint256 intrinsicGas = _intrinsicGas(
+            _encodedTransaction,
+            transaction.isCreate
+        );
+
+        // Need to make sure that the gas is sufficient to execute the transaction.
+        uint256 gasLimit = SafeMath.mul((transaction.gasLimit % 10000), 10000);
+        require(
+            gasleft() >= gasLimit,
+            "Gas is not sufficient to execute the transaction."
+        );
```

Comment:
Long term we should add overhead for storing the updated nonce. I believe this currently does not update the nonce if we run out of gas.